### PR TITLE
Issue/stautdef not initialized

### PR DESCRIPTION
**Pull request by pjljvandelaar**
removed constraint that state variables must be initialized.

ensured that TorXakis will not crashed when a variable is used before be assigned a value.

### CODE REVIEW

File: examps/Fibonacci/fibonacci.txs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{-
```
Comment (capitanbatata) [inline examps/Fibonacci/fibonacci.txs]: Shouldn't we have a convention regarding capitalization of files for the sake of consistency?

File: sys/behave/src/Expand.hs [hunk 2]
```diff
@@ -106,20 +117,24 @@ expand chsets (BNbexpr we (Parallel chans bexps))  =
...
      evalChanOffer _ (Quest vid) =
-          return $ Quest vid
+          return $ Right (Quest vid)
 
      evalChanOffer we' (Exclam vexp)  =  do
           tds <- gets IOB.tdefs
-          wal <- Eval.eval $ subst (Map.map cstrConst we') (funcDefs tds) vexp
-          return $ Exclam (cstrConst wal)
-
+          case TxsDefs.eval $ subst (Map.map cstrConst we') (funcDefs tds) vexp of
```
Comment (capitanbatata) [inline sys/behave/src/Expand.hs]: Why not use `right` from `Control.Arrow`:

```haskell
let res = TxsDefs.eval $ subst (Map.map cstrConst we') (funcDefs tds) vexp of
-- right :: a b c -> a (Either d b) (Either d c) 
return $ right (Exclam . cstrConst) res
```

File: sys/behave/src/Expand.hs [hunk 3]
```diff
@@ -137,9 +152,13 @@ expand chsets (BNbexpr we (ProcInst procid chans vexps))  =  do
      case Map.lookup procid (procDefs tdefs) of
        Just (ProcDef chids vids bexp)
          -> do let chanmap = Map.fromList (zip chids chans)
-               wals  <- mapM (Eval.eval . subst (Map.map cstrConst we) (funcDefs tdefs) ) vexps
-               let we' = Map.fromList (zip vids wals)
-               expand chsets $ BNbexpr we' (relabel chanmap bexp)
+               let wals = map (TxsDefs.eval . subst (Map.map cstrConst we) (funcDefs tdefs) ) vexps
+               case Data.Either.partitionEithers wals of
```
Comment (capitanbatata) [inline sys/behave/src/Expand.hs]: I'm thinking that, if we want to make this code transferable to anybody maybe it is a good idea to add some comments about what's going on here. Why are we partitioning the `Either`'s? What does each case represent? (for instance what does it mean to have no lefts?)

**Comment by capitanbatata** [comment]
The tests for `ANY` are failing, could you check that @pjljvandelaar?

File: examps/Fibonacci/fibonacci.txs (hunk 1 above)
Comment (pjljvandelaar) [reply examps/Fibonacci/fibonacci.txs]: If we check it, I  am fine with it!

File: sys/behave/src/Expand.hs (hunk 3 above)
Comment (pjljvandelaar) [reply sys/behave/src/Expand.hs]: Isn't that clear from what Eval does?

File: sys/behave/src/Expand.hs (hunk 2 above)
Comment (pjljvandelaar) [reply sys/behave/src/Expand.hs]: signature is incomprehensible to me.
Please explain it to me next Monday face 2 face!

**Comment by pjljvandelaar** [comment]
I propose to squash and merge.
Make a TODO / issue of using right!

**Pull request merged**